packages/amplify_datastore_plugin_interface/lib/src/types/hub/SubscriptionDataProcessedEvent.dart

```diff
@@ -0,0 +1,28 @@
+/*
+ * Copyright 2022 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+import 'package:amplify_datastore_plugin_interface/amplify_datastore_plugin_interface.dart';
+import 'package:amplify_core/amplify_core.dart';
+
+class SubscriptionDataProcessedEvent<M extends Model> extends HubEventPayload {
+  late HubEventElementWithMetadata<M> element;
+  late String modelName;
+
+  SubscriptionDataProcessedEvent(Map<dynamic, dynamic> serializedData,
+      ModelProviderInterface provider, String eventName) {
+    element = HubEventElementWithMetadata.fromMap(serializedData, provider);
+    modelName = serializedData["modelName"] as String;
+  }
+}
```

Comment:
```suggestion
class SubscriptionDataProcessedEvent<M extends Model> extends HubEventPayload {
  final HubEventElementWithMetadata<M> element;
  final String modelName;

  SubscriptionDataProcessedEvent(this.element, this.modelName);

  factory SubscriptionDataProcessedEvent.fromMap(
    Map<dynamic, dynamic> serializedData,
    ModelProviderInterface provider,
    String eventName,
  ) {
    final element =
        HubEventElementWithMetadata<M>.fromMap(serializedData, provider);
    final modelName = serializedData["modelName"] as String;
    return SubscriptionDataProcessedEvent(element, modelName);
  }
}
```

Reply:
I like this version better, but I will keep the current implementation following other event class implementations for now.